sway-lib-std/src/flags.sw

```diff
@@ -30,3 +89,196 @@ pub fn enable_panic_on_overflow() {
+///
+/// ### Examples
+///
+/// ```sway
+/// use std::{assert::assert, flags::{disable_panic_on_unsafe_math, enable_panic_on_unsafe_math}, registers::error};
+///
+/// fn main() {
+///     disable_panic_on_unsafe_math();
+///      
+///     let bar = 1 / 0; // Division by zero is considered unsafe math.
+///     assert(error() == 1); // Error flag is set to true whenever unsafe math occurs.
+///
+///     enable_panic_on_unsafe_math();
+/// }
+/// ```
+pub fn disable_panic_on_unsafe_math() {
+    // Mask first bit, which is `F_UNSAFEMATH`.
+    let mask = 0b00000000_00000000_00000000_00000000_00000000_00000000_00000000_00000001;
+
+    // Get the current value of the flags register and mask it, setting the
+    // masked bit. Flags are inverted, so set = off.
+    let flag_val = flags() | mask;
+    asm(flag_val: flag_val) {
+        flag flag_val;
+    }
+}
+
+/// Enables the default `panic-on-unsafe-math` behavior in the FuelVM.
+///
+/// > **_Note:_**
+/// >
+/// > `panic-on-unsafe-math` is the default, so there is no need to use this function unless you have previously called `disable_panic_on_unsafe_math`.
+///
+/// ### Examples
+///
+/// ```sway
+/// use std::{assert::assert, flags::{disable_panic_on_unsafe_math, enable_panic_on_unsafe_math}, registers::error};
+///
+/// fn main() {
+///     disable_panic_on_unsafe_math();
+///      
+///     let bar = 1 / 0; // Division by zero is considered unsafe math.
+///     assert(error() == 1); // Error flag is set to true whenever unsafe math occurs.
+///
+///     enable_panic_on_unsafe_math();
+/// }
+/// ```
+pub fn enable_panic_on_unsafe_math() {
+    // Mask first bit, which is `F_UNSAFEMATH`.
+    let mask = 0b11111111_11111111_11111111_11111111_11111111_11111111_11111111_11111110;
+
+    // Get the current value of the flags register and mask it, unsetting the
+    // masked bit. Flags are inverted, so unset = on.
+    let flag_val = flags() & mask;
+    asm(flag_val: flag_val) {
+        flag flag_val;
+    }
+}
+
+/// Allows unsafe math operations to occur without a FuelVM panic.
+/// More suitable than `disable_panic_on_unsafe_math` for use in functions that are not the entry point of a program.
+///
+/// > **_WARNING:_**
+/// >
+/// > Don't forget to call `set_flags` after performing the operations for which you disabled the default `panic-on-unsafe-math` behavior in the first place!
+///
+/// ### Examples
+///
+/// ```sway
+/// use std::{assert::assert, flags::{disable_panic_on_unsafe_math_preserving, set_flags}, registers::error};
+///
+/// fn foo() {
+///     let prior_flags = disable_panic_on_unsafe_math_preserving();
+///      
+///     let bar = 1 / 0; // Division by zero is considered unsafe math.
+///     assert(error() == 1); // Error flag is set to true whenever unsafe math occurs.
+///
+///     set_flags(prior_flags);
+/// }
+/// ```
+pub fn disable_panic_on_unsafe_math_preserving() -> u64 {
+    let prior_flags = flags();
+
+    // Mask first bit, which is `F_UNSAFEMATH`.
```

Comment:
I think it would be worthwhile to set these to a `F_UNSAFEMATH` constant instead of defining them in the function itself